File: JavaProjects/java-lang/src/main/java/practice/javalang8/enums/Card.java
package practice.javalang8.enums;

import java.util.ArrayList;
import java.util.List;

public class Card {

    private static final List<Card> protoDeck = new ArrayList<Card>();

    // Initialize prototype deck
    static {
        for (final Suit suit : Suit.values()) {
            for (final Rank rank : Rank.values()) {
                protoDeck.add(new Card(rank, suit));
            }
        }
    }

    private final Rank rank;
    private final Suit suit;

    private Card(final Rank rank, final Suit suit) {

        this.rank = rank;
        this.suit = suit;
    }

    public static ArrayList<Card> newDeck() {

        return new ArrayList<Card>(protoDeck); // Return copy of prototype deck
    }

    public Rank rank() {

        return rank;
    }

    public Suit suit() {

        return suit;
    }

    @Override
    public String toString() {

        return rank + " of " + suit;
    }

    public enum Rank {
        DEUCE, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING, ACE
    }

    public enum Suit {
        CLUBS, DIAMONDS, HEARTS, SPADES
    }
}

File: PythonProjects/python-kata/src/com/py/kata/suits/Card.py
class Cards(object):
    def __init__(self, suit=0, rank=2):
        if suit == 1 or rank == 12:
            print("Queen of Diamonds")
        self.suit = suit
        self.rank = rank

    def card_play(self):
        print("Suit:" + str(self.suit))
        print("Rank:" + str(self.rank))


def main_method():
    c = Cards(1, 12)
    c.card_play()


main_method()
